### PR TITLE
Code from private repository

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: engine/dlib/src/dlib/socket_posix.cpp [hunk 1]
```diff
@@ -83,11 +83,11 @@ namespace dmSocket
             DM_SOCKET_NATIVE_TO_RESULT_CASE(ADDRINUSE);
             DM_SOCKET_NATIVE_TO_RESULT_CASE(CONNABORTED);
             DM_SOCKET_NATIVE_TO_RESULT_CASE(INPROGRESS);
+        default:
```
Comment (JCash) [inline engine/dlib/src/dlib/socket_posix.cpp:86]: Warning fix

File: engine/dlib/src/dlib/endian_posix.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// Copyright 2020-2022 The Defold Foundation
```
Comment (JCash) [inline engine/dlib/src/dlib/endian_posix.h:1]: Easier to keep platforms apart, by putting it in a "platform" file

File: engine/dlib/src/wscript [hunk 3]
```diff
@@ -63,54 +56,46 @@ def build(bld):
...
-
     source_files = bld.path.ant_glob(source_dirs)
 
+    platform_source_files = []
+    if platform_source_dirs:
+        platform_source_files = bld.path.ant_glob(platform_source_dirs)
+
+    # Trying to maintain this set of rules in a way that fits all platforms+combinations is tricky
+    # The current incarnation is add all common sources plus the sources for the particular platform
+    # We then remove some specific files for each platform
```
Comment (JCash) [inline engine/dlib/src/wscript:67]: Tried to simplify the private platform support, by being much more explicit, and using less "magic" code.

File: scripts/build.py [hunk 4]
```diff
@@ -139,7 +139,8 @@ def get_target_platforms():
     if 'WD' in os.environ:
         SHELL= '%s\\bash.exe' % os.environ['WD'] # the binary directory
 
-ENGINE_LIBS = "testmain ddf particle glfw graphics lua hid input physics resource extension script render rig gameobject gui sound liveupdate crash gamesys tools record iap push iac webview profiler facebook engine sdk".split()
+ENGINE_LIBS = "testmain dlib texc ddf particle glfw graphics lua hid input physics resource extension script render rig gameobject gui sound liveupdate crash gamesys tools record iap push iac webview profiler facebook engine sdk".split()
+HOST_LIBS = "testmain dlib texc modelc".split()
```
Comment (JCash) [inline scripts/build.py:143]: Introduced a new variable, to highlight what's being built on the host platforms

File: scripts/build.py [hunk 5]
```diff
@@ -2009,6 +2012,7 @@ def _form_env(self):
                       '%s/build_tools' % self.defold,
                       '%s/ext/lib/python' % self.dynamo_home]
         env['PYTHONPATH'] = os.path.pathsep.join(pythonpaths)
+        env['PYTHONIOENCODING'] = 'UTF-8'
```
Comment (JCash) [inline scripts/build.py:2015]: I don't actually remember why this was needed. Can be related to older environment where I used msys.

File: engine/dlib/src/wscript [hunk 6]
```diff
@@ -63,54 +56,46 @@ def build(bld):
...
+
+    def remove_files(sources, patterns):
+        def matches(path, patterns):
+            for p in patterns:
+                if path.endswith(p):
+                    return True
+            return False
+        return [x for x in sources if not matches(x.abspath(), patterns)]
```
Comment (Jhonnyg) [inline engine/dlib/src/wscript:75]: 👍

**Review by Jhonnyg — APPROVED** [review]
Looks good!

**Pull request merged**